app/Resources/assets/js/jquery.instantSearch.js

```diff
@@ -0,0 +1,79 @@
+// jQuery plugin for an instant searching
+(function($) {
+    $.fn.instantSearch = function(config) {
+        return this.each(function() {
+            initInstantSearch(this, $.extend(true, defaultConfig, config || {}));
+        });
+    };
+
+    var defaultConfig = {
+        minQueryLength: 2,
+        maxPreviewItems: 10,
+        previewDelay: 500,
+        noItemsFoundMessage: 'No items found'
+    };
+
+    var throttle = (function(){
+        var timer = 0;
+        return function(callback, ms){
+            clearTimeout (timer);
+            timer = setTimeout(callback, ms);
+        };
+    })();
+
+    var initInstantSearch = function(el, config) {
+        var $input = $(el);
+        var $form = $input.parents('form').first();
+        var $preview = $('<ul class="search-preview list-group"></ul>').appendTo($form);
+
+        var setPreviewItems = function(items) {
+            $preview.empty();
+
+            $.each(items, function(index, item) {
+                if (index > config.maxPreviewItems) {
+                    return;
+                }
+
+                addItemToPreview(item)
+            });
+        }
+
+        var addItemToPreview = function(item) {
+            $preview.append('<li class="list-group-item"><a href="' + item.url + '">' + item.result + '</a></li>');
```

Comment:
this still has the XSS hole for item.result and item.url here. Here would be the proper way to do it:

```js
var $link = $('<a>').attr('href', item.url).text(item.result);
var $li = $('<li class="list-group-item">').append($link);

$preview.append($li);
```

This would let the browser handle the escaping of the text and prevent the ability to set extra thinks through the URL by having ``#" onclick="alert('xss')`` in the URL